# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/ActorsParent.cpp
# Commit: 1ae9231cee22
# Full Hash: 1ae9231cee224fc50ea5c1470a27b8cf1a1dbe9a
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-15 21:41:57
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1924658 - Downgrade an assertion in QuotaManager::GetOrCreateTemporaryOriginDirectory; r=dom-storage-reviewers,asuth
#   
#   The new diagnostic assertion has been added in bug 1866402 and it revealed a
#   long standing edge case. This patch downgrades the assertion into a debug only
#   assertion until the edge case is covered in a dedicated bug 1924680.
# ==============================================================================

diff -r d4e29a519c0d -r 1ae9231cee22 dom/quota/ActorsParent.cpp
--- a/dom/quota/ActorsParent.cpp	Tue Oct 15 16:45:15 2024 +0000
+++ b/dom/quota/ActorsParent.cpp	Tue Oct 15 16:54:35 2024 +0000
@@ -3215,7 +3215,7 @@
   MOZ_ASSERT(aOriginMetadata.mPersistenceType != PERSISTENCE_TYPE_PERSISTENT);
   MOZ_DIAGNOSTIC_ASSERT(IsStorageInitializedInternal());
   MOZ_DIAGNOSTIC_ASSERT(IsTemporaryStorageInitializedInternal());
-  MOZ_DIAGNOSTIC_ASSERT(IsTemporaryOriginInitializedInternal(aOriginMetadata));
+  MOZ_ASSERT(IsTemporaryOriginInitializedInternal(aOriginMetadata));
 
   QM_TRY_UNWRAP(auto directory, GetOriginDirectory(aOriginMetadata));
 
